tests/keras/backend/backend_test.py

```diff
@@ -44,10 +43,12 @@
 
 
 def check_dtype(var, dtype):
-    if K.backend() == 'theano':
-        assert var.dtype == dtype
-    else:
+    if K.backend() == 'tensorflow':
         assert var.dtype.name == '%s_ref' % dtype
```

Comment:
`dtype in str(var.dtype)` like in CNTK would seem like a better check. Dtypes in TF are weird, and both `float32_ref` and `float32` are possible values for the `name` attribute of a float32 variable.

Reply:
@fchollet, Updated.